core/src/buffer.rs

```diff
@@ -37,6 +37,13 @@ use scale_info::{
 #[derive(Clone, Default, Eq, Hash, Ord, PartialEq, PartialOrd, Decode, Encode, TypeInfo)]
 pub struct LimitedVec<T, E, const N: usize>(Vec<T>, PhantomData<E>);
 
+impl<T, E, const M: usize, const N: usize> From<[T; M]> for LimitedVec<T, E, N> {
+    fn from(value: [T; M]) -> Self {
+        assert!(M <= N);
+        Self(value.into(), Default::default())
+    }
+}
+
```

Comment:
```suggestion
struct LessThenOrEqual<const M: usize, const N: usize> {}

impl<const M: usize, const N: usize> LessThenOrEqual<M, N> {
    const ASSERTION: () = assert!(M <= N);
}

impl<T, E, const M: usize, const N: usize> From<[T; M]> for LimitedVec<T, E, N> {
    fn from(value: [T; M]) -> Self {
        let _ = LessThenOrEqual::<M, N>::ASSERTION;
        Self(value.into(), Default::default())
    }
}
```

Reply:
please also replace places where `::from` can be used

Reply:
this is still runtime assertion isnt it?

Reply:
> this is still runtime assertion isnt it?

No